### PR TITLE
Makefile libzmq

### PR DESCRIPTION
**Pull request by rezahousseini**
Adding new library "libzmq"

### CODE REVIEW

**Comment by starius** [comment]
Please squash two commits (7cd37e8 and be920fb). Adding a package is an atomic change.

File: src/libzmq.mk [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+$(PKG)_VERSION  := 4.1.3
+$(PKG)_CHECKSUM := 61b31c830db377777e417235a24d3660a4bcc3f40d303ee58df082fcd68bf411
+$(PKG)_SUBDIR   := zeromq-$($(PKG)_VERSION)
+$(PKG)_FILE     := zeromq-$($(PKG)_VERSION).tar.gz
+$(PKG)_URL      := http://download.zeromq.org/$($(PKG)_FILE)
+$(PKG)_DEPS     := gcc libsodium
+
+define $(PKG)_BUILD
+    cd '$(1)' && CHOST='$(TARGET)' ./configure \
+        --prefix='$(PREFIX)/$(TARGET)' \
```
Comment (starius) [inline src/libzmq.mk]: Please use $(MXE_CONFIGURE_OPTS).

**Comment by starius** [comment]
Please provide automatic updater of the package. See other `*.mk` files (section `$(PKG)_UPDATE`) to get examples.

File: index.html [hunk 2]
```diff
@@ -2577,7 +2581,7 @@ <h2 id="packages">List of Packages</h2>
     <tr>
         <td class="package">yasm</td>
         <td class="website"><a href="http://yasm.tortall.net//">Yasm</a></td>
-    </tr>
+    </tr>
```
Comment (tonytheodore) [inline index.html]: Whitespace

File: src/libzmq.mk [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    grep -v rc | \
+    tail -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(1)' && CHOST='$(TARGET)' ./configure \
+        $(MXE_CONFIGURE_OPTS) \
+        --prefix='$(PREFIX)/$(TARGET)' \
+        PKG_CONFIG='$(PREFIX)/bin/$(TARGET)-pkg-config' \
+        --without-libsodium
```
Comment (tonytheodore) [inline src/libzmq.mk]: ?? I'm guessing this means something like "don't build internal libsodium"

**Comment by tonytheodore** [comment]
Static builds are apparently disabled:
```
configure: error: Building static libraries is not supported under MinGW32
```

File: src/libzmq.mk [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    $(WGET) -q -O- 'http://download.zeromq.org/' | \
+    $(SED) -n 's,.*zeromq-\([0-9][^>]*\)\.tar.*,\1,p' | \
+    grep -v alpha | \
+    grep -v beta | \
+    grep -v rc | \
+    tail -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(1)' && CHOST='$(TARGET)' ./configure \
```
Comment (tonytheodore) [inline src/libzmq.mk]: CHOST isn't used

File: src/libzmq.mk [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        $(MXE_CONFIGURE_OPTS) \
+        --prefix='$(PREFIX)/$(TARGET)' \
```
Comment (tonytheodore) [inline src/libzmq.mk]: `--prefix` is set by `$(MXE_CONFIGURE_OPTS)`

File: src/libzmq.mk (hunk 3 above)
Comment (tonytheodore) [inline src/libzmq.mk]: Should be `--with-libsodium`

**Comment by tonytheodore** [comment]
Added in https://github.com/mxe/mxe/commit/90fc2216c24ccb32a1933c17cd3df3b346c8d792

**Pull request closed without merging**